File: src/main/kotlin/org/tendiwa/plane/geometry/straightSkeleton/MutableFace.java
package org.tendiwa.plane.geometry.straightSkeleton;


interface MutableFace extends Iterable<Node> {
    void addLink(Node one, Node another);

    boolean isClosed();

    StraightSkeletonFace toPolygon();

    Node getNodeFromLeft(LeftSplitNode leftNode);

    Node getNodeFromRight(RightSplitNode rightNode);
}

File: src/main/kotlin/org/tendiwa/plane/geometry/straightSkeleton/MutableFace.kt
package org.tendiwa.plane.geometry.straightSkeleton


internal interface MutableFace : Iterable<Node> {
    fun addLink(one: Node, another: Node)

    val isClosed: Boolean

    fun toPolygon(): StraightSkeletonFace

    fun getNodeFromLeft(leftNode: LeftSplitNode): Node

    fun getNodeFromRight(rightNode: RightSplitNode): Node
}
